cognite/client/_api/organization.py

```diff
@@ -35,3 +42,145 @@ def _organization(self) -> str:
                 "Could not look-up organization", response.status_code, response.headers.get("x-request-id")
             )
         return response.json()["organization"]
+
+
+class PrincipalsAPI(OrgAPI):
+    _RESOURCE_PATH = "/principals"
+
+    def me(self) -> Principal:
+        """`Get the current caller's information. <https://developer.cognite.com/api#tag/Principals/operation/getMe>`_
+
+        Returns:
+            Principal: The principal of the user running the code, i.e. the
+                principal *this* CogniteClient was instantiated with.
+
+        Examples:
+            Get your own principal:
+                >>> from cognite.client import CogniteClient
+                >>> client = CogniteClient()
+                >>> res = client.iam.principals.me()
+        """
+        # the /me endpoint is not using the /orgs/{org} base path, so we have to construct the URL manually
+        path = "/principals/me"
+        if self._api_version:
+            path = f"/api/{self._api_version}{path}"
+        full_url = urljoin(self._auth_url, path)
+        headers = self._configure_headers(
+            "application/json",
+            additional_headers=self._config.headers.copy(),
+            api_subversion=self._api_subversion,
+        )
+        response = self._http_client_with_retry.request(method="GET", url=full_url, headers=headers)
+        response.raise_for_status()
+        return Principal._load(response.json())
+
+    @overload
+    def retrieve(
+        self,
+        id: str,
+        *,
+        ignore_unknown_ids: bool = False,
+    ) -> Principal | None: ...
+
+    @overload
+    def retrieve(
+        self,
+        *,
+        external_id: str,
+        ignore_unknown_ids: bool = False,
+    ) -> Principal | None: ...
+
+    @overload
+    def retrieve(
+        self,
+        id: SequenceNotStr[str],
+        *,
+        ignore_unknown_ids: bool = False,
+    ) -> PrincipalList: ...
+
+    @overload
+    def retrieve(
+        self,
+        *,
+        external_id: SequenceNotStr[str],
+        ignore_unknown_ids: bool = False,
+    ) -> PrincipalList: ...
+
+    @overload
+    def retrieve(
+        self,
+        id: None = None,
+        external_id: None = None,
+        ignore_unknown_ids: bool = False,
+    ) -> PrincipalList: ...
```

Comment:
Is there any point in having this overload if we don't allow id and external_id to be provided at the same time?
```suggestion
```